src/java.base/unix/classes/sun/nio/fs/UnixFileSystem.java

```diff
@@ -1037,6 +1045,10 @@ void copy(final UnixPath source,
             if (!flags.replaceExisting)
                 throw new FileAlreadyExistsException(
                     target.getPathForExceptionMessage());
+            else if (!sourceAttrs.isSymbolicLink() || flags.followLinks)
+                // Ensure source can be read
+                provider.checkAccess(source, AccessMode.READ);
```

Comment:
I think access(source, R_OK) would be better here, otherwise the additional SM check is observable.

Reply:
An oversight; thanks.

Reply:
Fixed in e226096d15b47bee7cee20ec2994bca93a071433.